server/src/main/java/org/opensearch/cluster/ApplicationManager.java

```diff
@@ -0,0 +1,153 @@
+
+        if (extensionManager.getExtensionPrincipals().contains(principal)) {
+            return extensionManager.getExtensionIdMap()
+                .get(principal.getName())
+                .getScopes()
+                .stream()
+                .filter(scope -> Scope.parseScopeFromString(scope).getNamespace() == ScopeEnums.ScopeNamespace.APPLICATION)
+                .collect(Collectors.toSet());
+        }
+
+        return Set.of();
+    }
+
+    /**
+     * This method allows for checking the ActionScopes associated with an application
+     * @param principal The principal of the application whose scopes you want to check
+     * @return A set of all action scopes associated with that principal
+     */
+    public Set<String> getActionScopes(Principal principal) {
+
+        if (extensionManager.getExtensionPrincipals().contains(principal)) {
+            return extensionManager.getExtensionIdMap()
+                .get(principal.getName())
+                .getScopes()
+                .stream()
+                .filter(scope -> Scope.parseScopeFromString(scope).getNamespace() == ScopeEnums.ScopeNamespace.ACTION)
+                .collect(Collectors.toSet());
+        }
+
+        return Set.of();
+    }
+
+    /**
+     * Checks scopes of an application subject and determine if it is allowed to perform an operation based on the given scopes
+     * @param wrapped The ApplicationSubject whose scopes should be evaluated
+     * @param scopes The scopes to check against the subject
+     * @return true if allowed, false if none of the scopes are allowed.
+     */
+    public boolean isAllowed(ApplicationSubject wrapped, final List<Scope> scopes) {
+
+        final Optional<Principal> optionalPrincipal = wrapped.getApplication();
+
+        if (optionalPrincipal.isEmpty()) {
+            // If there is no application, actions are allowed by default
+            return true;
+        }
+
+        if (!wrapped.applicationExists()) {
+            return false;
+        }
+
+        final Set<String> scopesOfApplication = wrapped.getScopes();
+
+        boolean isApplicationSuperUser = scopesOfApplication.stream()
+            .map(Scope::parseScopeFromString)
+            .anyMatch(parsedScope -> parsedScope.equals(ApplicationScope.SUPER_USER_ACCESS));
+
+        if (isApplicationSuperUser) {
+
+            return true;
+        }
+
+        boolean isMatchingScopePresent = scopesOfApplication.stream()
+            .map(Scope::parseScopeFromString)
+            .anyMatch(parsedScope -> scopes.stream().anyMatch(parsedScope::equals));
+
+        return isMatchingScopePresent;
+    }
+
+    /**
+     * This method allows for checking the ExtensionPointScopes associated with an application
+     * @param principal The principal of the application whose scopes you want to check
+     * @return A set of all extension point scopes associated with that principal
+     */
+    public Set<String> getExtensionPointScopes(Principal principal) {
```

Comment:
The method is named `getExtensionPointScopes`, why doesn't it return scope objects?  Scopes have a method asPermissionString to transform them into a string if the caller wants that, right?

Reply:
This was being used to get the names like "Action.READ" etc. I can swap to just the full scope permission name for those cases and make this return scope objects.